### PR TITLE
Write initial implementation

### PR DESCRIPTION
**Pull request by forabi**

### CODE REVIEW

**Comment by forabi** [comment]
I think this is usable, although there are still some improvements to be made. Merging to test CI pipeline and functionality in live Lambda environment.

File: src/runReporters.ts [hunk 1]
```diff
@@ -4,20 +4,156 @@
...
-  console.log(await bluebird.fromNode(cb => wpt.getTestResults(testId, cb)));
+  const rawReports = mapValues(keyBy(results, r => r.url), r => r.raw);
+  let markdownReport = stripIndents`
+    Report for tests performed on ${dateStr}
+    ========================================
 
-  callback(null, 'Hello World');
+    ${results
+      .map(({ url, rendered }) => {
+        return stripIndents`
```
Comment (wholesomedev) [inline src/runReporters.ts]: I've always found this syntax weird ``foo`some template string` ``. I'll need to read up on it eventually 😄

File: src/reporters/WebPageTestReporter.ts [hunk 2]
```diff
@@ -0,0 +1,164 @@
...
+          name: 'Lighthouse via WebPageTest',
+          records: WebPageTestReporter.getLighthouseRecords(
+            runTestResponse.data,
+          ),
+        });
+      } else {
+        // tslint:disable-next-line:no-multiline-string
+        console.warn(oneLine`
+          Could not find Lighthouse data in WPT API response.
+          Does the location specified support Lighthouse?
```
Comment (wholesomedev) [inline src/reporters/WebPageTestReporter.ts]: What is meant by `location specified` here?

Comment (forabi) [reply src/reporters/WebPageTestReporter.ts]: One of www.webpagetest.org/getLocations.php?f=html. I should add this to the message.

File: src/runReporters.ts [hunk 3]
```diff
@@ -4,20 +4,156 @@
...
+      rendered: renderReport({
+        reports,
+      }),
+    };
+  });
 
-  // tslint:disable-next-line:no-console
-  console.log(await bluebird.fromNode(cb => wpt.getTestResults(testId, cb)));
+  const rawReports = mapValues(keyBy(results, r => r.url), r => r.raw);
+  let markdownReport = stripIndents`
```
[the remark this answers is not in the document]
Comment (forabi) [reply src/runReporters.ts]: Yes, because leading white space is significant in markdown. Prettier cannot remove it because it would change the output.

**Pull request merged**